OPAL/br/src/main/scala/org/opalj/br/fpcf/properties/fieldaccess/FieldAccessInformation.scala

```diff
@@ -0,0 +1,114 @@
+
+    def readAccesses(implicit declaredMethods: DeclaredMethods): Iterator[(DefinedMethod, PC)] =
+        encodedReadAccesses.iterator.map(decodeFieldAccess(_))
+
+    def writeAccesses(implicit declaredMethods: DeclaredMethods): Iterator[(DefinedMethod, PC)] =
+        encodedWriteAccesses.iterator.map(decodeFieldAccess(_))
+
+    def getNewestNReadAccesses(n: Int)(implicit declaredMethods: DeclaredMethods): Iterator[(DefinedMethod, PC)] =
+        encodedReadAccesses.iterator.take(n).map(decodeFieldAccess(_))
+
+    def getNewestNWriteAccesses(n: Int)(implicit declaredMethods: DeclaredMethods): Iterator[(DefinedMethod, PC)] =
+        encodedWriteAccesses.iterator.take(n).map(decodeFieldAccess(_))
+
+    def numReadAccesses: Int = encodedReadAccesses.size
+    def numWriteAccesses: Int = encodedWriteAccesses.size
+
+    def included(other: FieldAccessInformation): FieldAccessInformation = included(other, 0, 0)
+    def included(
+        other:             FieldAccessInformation,
+        seenReadAccesses:  Int,
+        seenWriteAccesses: Int
+    ): FieldAccessInformation = {
+        var newReadAccesses = encodedReadAccesses
+        other.encodedReadAccesses.forFirstN(other.numReadAccesses - seenReadAccesses)(newReadAccesses += _)
+        var newWriteAccesses = encodedWriteAccesses
+        other.encodedWriteAccesses.forFirstN(other.numWriteAccesses - seenWriteAccesses)(newWriteAccesses += _)
+
+        if (newReadAccesses == encodedReadAccesses && newWriteAccesses == encodedWriteAccesses)
+            return this;
+
+        FieldAccessInformation(newReadAccesses, newWriteAccesses)
+    }
+}
+
+object FieldAccessInformation extends FieldAccessInformationPropertyMetaInformation {
+
+    /**
+     * The key associated with every field access property. The name is "FieldAccessInformation"; the fallback is
+     * computed via the fallback reason to ensure an analysis for the information was scheduled.
+     */
+    final val key = PropertyKey.create[Field, FieldAccessInformation](
+        "FieldAccessInformation",
+        (_: PropertyStore, reason: FallbackReason, _: Entity) =>
+            reason match {
+                case PropertyIsNotDerivedByPreviouslyExecutedAnalysis => NoFieldAccessInformation
+                case _ =>
+                    throw new IllegalStateException("No analysis is scheduled for property FieldAccessInformation")
+            }
+    )
+
+    def apply(
+        readAccesses:  Set[(DefinedMethod, PCs)],
+        writeAccesses: Set[(DefinedMethod, PCs)]
+    ): FieldAccessInformation = {
+        def getEncodedAccessSet(accesses: Set[(DefinedMethod, PCs)]): Set[FieldAccess] =
+            accesses.flatMap(methodPCs =>
+                methodPCs._2.foldLeft(Set.empty[FieldAccess])(_ + encodeFieldAccess(methodPCs._1, _)))
+
+        val encodedReadAccesses = getEncodedAccessSet(readAccesses).foldLeft(LongLinkedTrieSet.empty)(_ + _)
```

Comment:
Do the encoding and conversion to LongLinkedTrieSet in one step to avoid the additional intermediate set.